scripts/check_catalog_urls.py

```
@@ -0,0 +1,150 @@
+    last_err = None
+    for attempt in range(max_retries):
+        try:
+            with urllib.request.urlopen(req, timeout=timeout) as r:
+                return r.status, None
+        except urllib.error.HTTPError as e:
+            if e.code == 429 and attempt < max_retries - 1:
+                retry_after = e.headers.get("Retry-After")
+                wait = float(retry_after) if retry_after else delay
+                time.sleep(wait)
+                delay = min(delay * 2, 30.0)
+                last_err = f"HTTP 429 (retried)"
```

Ruff (0.15.11)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)